### PR TITLE
feat(nuxt): opt in to nuxt 4

### PR DESCRIPTION
**Pull request by antoinezanardi**

### CODE REVIEW

**Comment by antoinezanardi** [comment]
Hello @danielroe,

I believe this PR might interest you.

Here is a summary of the steps I followed to opt-in to the Nuxt 4 experimental features and get everything working:

1. Moved all source files to the `app` folder. I also moved `stores/`, even though it wasn't specified in the migration guide. However, I kept the `tests` folder in the root directory. I think these two examples should be mentioned in the guide for clarity.
2. Updated the path aliases in `tsconfig.json`, `config/vitest/vitest.unit-config.ts`, and `nuxt.config.ts`. Since the sources are now in the `app` folder, but other directories like `modules` are not, I had to set the path aliases as follows:

```json
"paths": {
  "~/*": ["app/*"],
  "@nuxt/*": [".nuxt/*"],
  "#components": [".nuxt/components"],
  "@tests/*": ["./tests/*"],
  "@modules/*": ["./modules/*"]
}
```

However, something peculiar happened. I needed to specify `~/*`, `@nuxt/*`, and `#components` again. Without these, my test files couldn't resolve the sources. Am I missing something here?

3. 🚨 **Possible Issue**: In the `app/composables/route/useWerewolvesAssistantRoutes.ts` file, the `useRoute` composable lost its type and is now `any`. I had to create a temporary type and disable the eslint `no-any` rule for now.

In summary, the migration was almost seamless. The main challenge was handling the path aliases, likely due to my unfamiliarity with them. Since the migration guide suggests moving all sources, it might be helpful to include information on updating path aliases in TypeScript as well.

Thank you for your attention and keep up the good work with the Nuxt team 💚 !!
